craft_providers/base.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Base configuration module."""
+import logging
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from .executor import Executor
+
+logger = logging.getLogger(__name__)
+
+
+class Base(ABC):
```

Comment:
can this be its own PR? It would have the benefit of being independent from the other ones too